src/map/clif.cpp

```diff
@@ -24693,6 +24699,29 @@ void clif_parse_itempackage_select( int fd, struct map_session_data* sd ){
 #endif
 }
 
+void clif_broadcast_refine_result(map_session_data& sd, t_itemid itemId, int8 level, bool success)
+{
+#if PACKETVER_MAIN_NUM >= 20170906 || PACKETVER_RE_NUM >= 20170830 || defined(PACKETVER_ZERO)
+	PACKET_ZC_BROADCAST_ITEMREFINING_RESULT packet{};
+	packet.packetType = HEADER_ZC_BROADCAST_ITEMREFINING_RESULT;
+	packet.itemId = itemId;
+	packet.refine_level = level;
+	packet.status = (int8)success;
+
+	if (battle_config.broadcast_hide_name) {
+		std::string dispname = std::string(sd.status.name);
+		int hide = min(battle_config.broadcast_hide_name, dispname.length() - 1);
+		dispname.replace(dispname.length() - hide, hide, hide, '*');
+		safestrncpy(packet.name, dispname.c_str(), NAME_LENGTH);
+	}
+	else {
+		safestrncpy(packet.name, sd.status.name, NAME_LENGTH);
+	}
```

Comment:
Can we put this in a separate function?
As this is used in clif_broadcast_obtain_special_item and here (and maybe should also be used in grade announce)?

Reply:
kRO does not hide the player's name. It could be an oversight, but I'm keeping it as is for now.